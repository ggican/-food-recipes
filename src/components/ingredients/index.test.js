import React from "react";
import Ingredients from "./index";

let successElement;
let failedElement;

beforeEach(() => {
    successElement = (
        <Ingredients title="Cooking" date="10-11-12"></Ingredients>
    );
    failedElement = <Ingredients></Ingredients>;
});

describe("Igredients Component", () => {
    it("SUCCESS renders and create snapshot  <Ingredients />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("SUCCESS renders get value title <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "text-ingredients-top" })
                .find(".ingredients__title")
                .text(),
        ).toEqual("Cooking");
    });

    it("SUCCESS renders get value date <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("10-11-12");
    });

    it("FAILED renders and create snapshot <Ingredients />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("FAILED renders get value title <Ingredients />", () => {
        expect(
            shallow(failedElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("-");
    });

    it("FAILED renders get value date <Ingredients />", () => {
        expect(
            shallow(failedElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("-");
    });
});
