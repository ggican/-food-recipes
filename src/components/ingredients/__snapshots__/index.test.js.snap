// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Igredients Component FAILED renders and create snapshot <Ingredients /> 1`] = `
<Ingredients
  date="10-11-12"
  title="Cooking"
>
  <styled.div
    className="ingredients"
  >
    <div
      className="sc-AxjAm beINzg ingredients"
    >
      <div
        className="ingredients--box"
      >
        <div
          className="ingredients__text"
          test_id="text-ingredients-top"
        >
          <div
            className="ingredients__caption"
          >
            Title
          </div>
          <div
            className="ingredients__title"
          >
            Cooking
          </div>
        </div>
        <div
          className="ingredients__text"
          test_id="text-ingredients-bottom"
        >
          <div
            className="ingredients__caption"
          >
            Use By
          </div>
          <div
            className="ingredients__date"
          >
            10-11-12
          </div>
        </div>
      </div>
    </div>
  </styled.div>
</Ingredients>
`;

exports[`Igredients Component SUCCESS renders and create snapshot  <Ingredients /> 1`] = `
<Ingredients
  date="10-11-12"
  title="Cooking"
>
  <styled.div
    className="ingredients"
  >
    <div
      className="sc-AxjAm beINzg ingredients"
    >
      <div
        className="ingredients--box"
      >
        <div
          className="ingredients__text"
          test_id="text-ingredients-top"
        >
          <div
            className="ingredients__caption"
          >
            Title
          </div>
          <div
            className="ingredients__title"
          >
            Cooking
          </div>
        </div>
        <div
          className="ingredients__text"
          test_id="text-ingredients-bottom"
        >
          <div
            className="ingredients__caption"
          >
            Use By
          </div>
          <div
            className="ingredients__date"
          >
            10-11-12
          </div>
        </div>
      </div>
    </div>
  </styled.div>
</Ingredients>
`;
